Cargo.toml

```diff
@@ -42,4 +42,4 @@ cxx-qt-lib-headers = { git = "https://github.com/kdab/cxx-qt.git", branch = "mai
 qt-build-utils = { git = "https://github.com/kdab/cxx-qt.git", branch = "main" }
 
 # We need a patched surfman to avoid OpenGL errors
-surfman = { git = "https://github.com/ahayzen-kdab/surfman.git", branch = "patched-opengl-assert-0-9-1" }
+surfman = { git = "https://github.com/ahayzen-kdab/surfman.git", branch = "patched-opengl-assert-and-gles-0-9-1" }
```

Comment:
Note the embedded world branch has this change and worked well :-) https://github.com/KDABLabs/cxx-qt-servo-webview/tree/demo-embedded-world

https://github.com/KDABLabs/cxx-qt-servo-webview/commit/0c76d94692ea98c4832cbfe497cde104617c09d8

Reply:
Yep :)
So we can close this PR, as forcing GLES on desktop is probably not wanted?

Otherwise it would be best to make upstream aware of this problem (or more likely fix it directly ourself).

Reply:
Maybe we create an issue on this repo for now? We had to hardcode surfman having it detect or have an option somewhere to pick GLES over GL on Linux is what we would need.

Reply:
So looking at [the patch](https://github.com/ahayzen-kdab/surfman/commit/47ceeda56e4b407adb1346911bd80b52ae871478), I actually don't think it would need much to fix this upstream.

I suppose instead of hardcoding a return value, instead look at the underlying EGL connection (for Wayland at least) and return the OpenGL version from there?

Reply:
> right it somehow needs to determine from the connection

Yes, but that's pretty straight forward with EGL, there is [eglQueryContext](https://registry.khronos.org/EGL/sdk/docs/man/html/eglQueryContext.xhtml) with the attribute `EGL_CONTEXT_CLIENT_TYPE`, which can return one of:
- `EGL_OPENGL_API`
- `EGL_OPENGL_ES_API`
- `EGL_OPENVG_API`

I suppose the following patch should work (for Wayland platform):
```diff
diff --git a/surfman/src/platform/unix/wayland/connection.rs b/surfman/src/platform/unix/wayland/connection.rs
index 5fdcbbf..6161de6 100644
--- a/surfman/src/platform/unix/wayland/connection.rs
+++ b/surfman/src/platform/unix/wayland/connection.rs
@@ -63,7 +63,27 @@ impl Connection {
     /// Returns the OpenGL API flavor that this connection supports (OpenGL or OpenGL ES).
     #[inline]
     pub fn gl_api(&self) -> GLApi {
-        GLApi::GL
+        unsafe {
+            EGL_FUNCTIONS.with(|egl| {
+                let display = egl.GetCurrentDisplay();
+                let context = egl.GetCurrentContext();
+                if context != egl::NO_CONTEXT {
+                    let mut client_type: i32 = 0;
+                    egl.QueryContext(
+                        display,
+                        context,
+                        egl::CONTEXT_CLIENT_TYPE as i32,
+                        &mut client_type,
+                    );
+                    match client_type as u32 {
+                        egl::OPENGL_ES_API => GLApi::GLES,
+                        _ => GLApi::GL,
+                    }
+                } else {
+                    GLApi::GL
+                }
+            })
+        }
     }
 
     /// Returns the "best" adapter on this system, preferring high-performance hardware adapters.
```

I can't test it right now unfortunately, as the Pi is not online anymore.